data/basics/userBalance.go

```diff
@@ -437,23 +437,35 @@ func PendingRewards(ot *OverflowTracker, proto config.ConsensusParams, microAlgo
 	return MicroAlgos{Raw: ot.Mul(rewardsUnits, rewardsDelta)}
 }
 
-// WithUpdatedRewards returns an updated number of algos in an AccountData
+// WithUpdatedRewards returns an updated number of algos, total rewards and new rewards base
 // to reflect rewards up to some rewards level.
-func (u AccountData) WithUpdatedRewards(proto config.ConsensusParams, rewardsLevel uint64) AccountData {
-	if u.Status != NotParticipating {
+func WithUpdatedRewards(
```

Comment:
I agree that the naming choice for this method is no longer correct, and it should be updated. I think that we can come up with something better the `UpdateRewards`, since the method doesn't really do any updating. But I'm going to defer that to a subsequent PR ( just because it's not really breaking anything as is ).